sys/src/9/amd64/l64vsyscall.S

```diff
@@ -73,15 +73,16 @@ syscallentry:
 	MOVQ	%rsp, %rsi 				/* Ureg* */
 	// system call number is in %rax, as per linux.
 	movq	%rax, %rdi
-	xorq	%rax, %rax
-	pushq	%rax
+	xorl	%eax, %eax
+	pushq	$2
 	popfq			/* clear all flags. is there something else we should clear too? */
-	movq	$0, %rbp	/* stack traces end here */
+	pushq	$0
```

Comment:
it might be best if this were a separate commit? What is this fixing?

Reply:
This was just wrong; removed.